tests/test_core.py

```diff
@@ -46,6 +48,22 @@ def test_device_unicode(self):
             for device in devices:
                 print("device: %s" % device)
 
+    def test_device_failed_properties(self):
+        """
+        Makes sure printing a device doesn't crash.
+        """
```

Comment:
This could potentially be a single line
https://www.python.org/dev/peps/pep-0257/#one-line-docstrings